src/Components/WithCurrentTime.tsx

```diff
@@ -36,6 +50,14 @@ export class WithCurrentTime extends React.Component<
     )
   }
 
+  async componentWillMount() {
+    if (this.props.useServerAdjustment) {
+      const timeOffsetInMilliseconds = await getOffsetBetweenGravityClock()
+
+      this.setState({ timeOffsetInMilliseconds })
```

Comment:
The offset could change over time. We should probably do this in a loop. Like, once a minute or so?

Reply:
Interesting. So, I wouldn't think it changes once the page loads, unless your system clock is counting seconds differently? or you change your clock _after_ the page loads.

Both of these feel like _not_ the normal user experience and an additional enhancement on top of this PR (which is already an enhancement over the current un-synced implementation), so while I think it's a good/interesting idea, I do feel less compelled to implement that in this PR.

Maybe we can note it as a future enhancement?

Reply:
> or you change your clock after the page loads.

Maybe not manual clock changes, but I'm thinking that unless you use a system from the 90s it's probably doing automatic time updates with some remote time server thing. So if it has ever drifted enough that we actually need to use an offset, then maybe it will fix itself soon and our offset will become wrong?

But yeah you're right that's unlikely to happen. Happy to leave it for now.

Reply:
Oh I see, yea good call on that possibility too!